### PR TITLE
Update developer guide (User Stories & Glossary)

### LINKED ISSUE #25: Update DG: User Stories

### LINKED ISSUE #26: Update DG: Glossary

### PR DESCRIPTION
**Pull request by riccoljy**
Closes #25 and closes #26

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -287,16 +287,45 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `*`      | user with many persons in the address book | sort persons by name         | locate a person easily                                                 |
-
-*{More to be added}*
+| Priority | As a …​         | I want to …​                                                                                                         | So that I can…​                                                                                                                                                    |
+|:--------:|-----------------|----------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `* * *`  | user            | add tag each contact based on created tags such as florist, musician etc                                             | I can easily understand the group this person belongs to                                                                                                           |
```
Comment (tingxuanp) [inline docs/DeveloperGuide.md:292]: Minor grammatical error but should it be just "tag each contact"?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -287,16 +287,45 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | user                                       | add a new person             |                                                                        |
-| `* * *`  | user                                       | delete a person              | remove entries that I no longer need                                   |
-| `* * *`  | user                                       | find a person by name        | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name         | locate a person easily                                                 |
-
-*{More to be added}*
+| Priority | As a …​         | I want to …​                                                                                                         | So that I can…​                                                                                                                                                    |
```
Comment (tingxuanp) [inline docs/DeveloperGuide.md:290]: The "I can" is repeated twice in most user stories when reading this way, perhaps you could change that?

**Review by tingxuanp — COMMENTED** [review]
LGTM except minor grammar errors.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (riccoljy) [reply docs/DeveloperGuide.md:292]: Didn't notice that -- thank you for catching that

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (riccoljy) [reply docs/DeveloperGuide.md:290]: I missed that out too; thank you for catching that!

**Pull request merged**